### PR TITLE
2398: Complete manual deploy workflow 2/2

### LINKED ISSUE #2398: Github workflow to deploy to another sandbox
### Issue description

In engineering huddle, we discussed having a new workflow that would let us to select a branch and deploy that branch to a sandbox of our choosing. The goal would be to easily move code between sandbox and not have to worry about collecting static locally and doing a cf push from our local env. This would help engineers that have multiple PRs in flight or who hit migrations issues on one sandbox.

### Acceptance criteria

- [x] a new github workflow is created
- [x] the github workflow let's a user select a branch
- [x] It also allows the user to select a sandbox (NOT including stable, staging or development)
- [x] upon running the workflow, the normal deploy steps (including compile, collect static and cf push) are all ran and the code is deployed to the correct sandbox.
- [x] the create dev sandbox script is updated to be sure new sandboxes get added to this workflow

### Additional context

**Important note:** workflows don't get applied until merging, expect that you will have multiple merges to review your action. Do not use the `resolves ticket #` language in your PR as that will close your ticket upon merging, instead put wording like `testing #` and get the dev reviewer to look at your github workflow on github itself. 

Because you need someone else to approve your PR this may work best pairing with someone, or branching the repo.

Really quick I assume the workflow needed would be: 
```
name: Manual Build and Deploy

on:
  workflow_dispatch:
    inputs:
      environment:
        description: 'Environment to deploy'
        required: true
        default: 'ab'
        type: choice
        options:
          - ab
          - ...# all the other sandboxes
          - meoward
          - bob
          - cb
      branch:
        description: 'Branch to deploy'
        required: true
        default: 'main'
        type: string

jobs:
  variables:
    runs-on: ubuntu-latest
    steps:
      - name: Setting global variables
        uses: actions/github-script@v6
        id: var
        with:
          script: |
            core.setOutput('environment', '${{ github.event.inputs.environment }}');
            core.setOutput('branch', '${{ github.event.inputs.branch }}');

  deploy:
```

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #2398 

## Changes

- Add any sandbox created using `create_dev_sandbox.sh` as a sandbox option in manual deploy workflow.
- Rename workflow from `deploy-branch-to-sandbox` to `deploy-mannal`. Note that this workflow will fail when run manually on GitHub actions until merged into main due to this name change.

## Context for reviewers

Completes remaining ACs for #2398. PR #2458 added the initial workflow to test and verify that it successfully deploys to sandbox.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [x] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: ops/scripts/create_dev_sandbox.sh [hunk 1]
```diff
@@ -112,10 +112,14 @@ sed -i '' '/          - development/ {a\
           - '"$1"'
 }' .github/workflows/reset-db.yaml
 
-sed -i '' '/          - development/ {a\
+sed -i '' '/          - backup/ {a\
```
Comment (abroddrick) [inline ops/scripts/create_dev_sandbox.sh:115]: **blocking change**  why did you change this? this tells it where to add stuff below, and we don't want to add it below developement, we want things added below development. please revert this.

File: ops/scripts/create_dev_sandbox.sh [hunk 2]
```diff
@@ -112,10 +112,14 @@ sed -i '' '/          - development/ {a\
...
           - '"$1"'
 }' .github/workflows/migrate.yaml
 
+sed -i '' '/          - development/ {a\
+          - '"$1"'
+}' .github/workflows/deploy-manual.yaml
```
Comment (abroddrick) [inline ops/scripts/create_dev_sandbox.sh:121]: @erinysong adding this to the create makes sense but you then have to make sure it gets removed in the destroy yaml

File: ops/scripts/create_dev_sandbox.sh (hunk 1 above)
Comment (erinysong) [reply ops/scripts/create_dev_sandbox.sh]: Sorry - meant to add new sandboxes under backup in the options in deploy-manual and not deploy. That was an oversight on my part so thank you for catching that!

File: docs/operations/README.md [hunk 3]
```diff
@@ -45,6 +45,8 @@ When deploying to your personal sandbox, you should make sure all of the USWDS a
 
 For ease of use, you can run the `deploy.sh <sandbox name>` script in the `/src` directory to build the assets and deploy to your sandbox. Similarly, you could run `build.sh <sandbox name>` script to just compile and collect the assets without deploying.
 
+You may also manually deploy to a sandbox using our [manual deploy workflow](https://github.com/cisagov/manage.get.gov/actions/workflows/deploy-manual.yaml) on GitHub Actions. Select Run workflow and enter the branch you want to deploy to your sandbox of choice.
```
Comment (erinysong) [inline docs/operations/README.md:48]: This link will only work once we merge since we're renaming our workflow from deploy-branch-to-sandbox to deploy-manual

File: ops/scripts/destroy_dev_sandbox.sh [hunk 4]
```diff
@@ -49,6 +49,7 @@ rm ops/manifests/manifest-$1.yaml
 sed -i '' "/getgov-$1.app.cloud.gov/d" src/registrar/config/settings.py
 sed -i '' "/- $1/d" .github/workflows/reset-db.yaml
 sed -i '' "/- $1/d" .github/workflows/migrate.yaml
+sed -i '' "/- $1/d" .github/workflows/deploy-manual.yaml
```
[the remark this answers is not in the document]
Comment (erinysong) [reply ops/scripts/destroy_dev_sandbox.sh:52]: Also am not super knowledgeable in sed but I believe the `d` command deletes the entire line so the whitespace in the bullet is also removed! (also did not know this until looking it up for this ticket)
